osc-server/src/main/java/org/osc/core/broker/service/BaseDeploymentSpecService.java

```diff
@@ -78,17 +84,64 @@ protected void validate(EntityManager em, DeploymentSpecDto dto) throws Exceptio
                 throw new VmidcBrokerValidationException("Invalid count " + dto.getCount()
                 + " specified for Deployment Specification. Only valid value is 1.");
             }
+            validateSelectionCriteria(this.vs, dto);
         }
     }
 
-    protected void throwInvalidUpdateActionException(String attributeType, String dsName)
+	private void validateSelectionCriteria(VirtualSystem vs, DeploymentSpecDto dto) throws Exception {
```

Comment:
can we use the ListAvailabilityZonesService and other services to validate instead of duplicating the code here?

Also, how about validation of other fields like tenant, floating ip, the networks etc